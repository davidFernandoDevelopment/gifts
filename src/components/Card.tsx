interface Props {
    id: number;
    url: string;
    title: string;
}

export const Card = ({ title, url }: Props) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};

export default Card;