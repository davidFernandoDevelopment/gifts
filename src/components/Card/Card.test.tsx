import { render, screen } from '@testing-library/react';
import { debug } from 'console';

import Card from './Card';


describe('Pruebas para Card', () => {
    const titleMock = 'Saitama';
    const urlMock = 'https://one-puch.com/saitama.png';


    test('Debe hacer match con el snapshot', () => {
        const {
            container
        } = render(<Card url={urlMock} title={titleMock} />);

        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la img con url indicado', () => {
        render(<Card url={urlMock} title={titleMock} />);

        const {
            src, alt
        } = screen.getByRole<HTMLImageElement>('img');

        expect(src).toBe(urlMock);
        expect(alt).toBe(titleMock);
    });

    test('Debe de mostrar el titulo del componente', () => {
        render(<Card title={titleMock} url={urlMock} />);

        expect(screen.getByText(titleMock)).toBeTruthy();
    });
});

