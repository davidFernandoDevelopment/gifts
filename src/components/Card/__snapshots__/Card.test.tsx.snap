// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para Card Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-puch.com/saitama.png"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
